rules/windows/process_creation/proc_creation_win_powershell_xor_commandline.yml

```diff
@@ -23,25 +32,20 @@ detection:
             - 'pwsh.dll'
         - Description: 'Windows PowerShell'
         - Product: 'PowerShell Core 6'
-    selection_cli:
+    selection_cli_xor:
+        CommandLine|contains: 'bxor'
+    selection_cli_other:
         CommandLine|contains:
-            - 'bxor'
+            - 'ForEach'
+            - 'for('
+            - 'for '
             - '-join '
-            - '-join'''
+            - "-join'"
```

Comment:
`'-join'''` is the yaml to escape `'` in a string.
https://yaml.org/spec/1.2.2/#57-escaped-characters
```
7.3.2. Single-Quoted Style

The single-quoted style is specified by surrounding “'” indicators. Therefore, within a single-quoted scalar, such characters need to be repeated. This is the only form of escaping performed in single-quoted scalars. In particular, the “\” and “"” characters may be freely used. This restricts single-quoted scalars to [printable](https://yaml.org/spec/1.2.2/#character-set) characters. In addition, it is only possible to break a long single-quoted line where a [space](https://yaml.org/spec/1.2.2/#white-space-characters) character is surrounded by non-[spaces](https://yaml.org/spec/1.2.2/#white-space-characters).
```

Reply:
the easy way is not always the answer